src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol_ImplementationChecks.cs

```diff
@@ -880,13 +880,32 @@ private void CheckNewModifier(Symbol symbol, bool isNew, DiagnosticBag diagnosti
                                 diagnostics.Add(ErrorCode.ERR_CantChangeRefReturnOnOverride, overridingMemberLocation, overridingMember, overriddenMember);
                                 suppressAccessors = true; //we get really unhelpful errors from the accessor if the ref kind is mismatched
                             }
-                            else if (!IsValidOverrideReturnType(overridingProperty, overridingMemberType, overriddenMemberType, diagnostics))
+                            else if (overridingProperty.SetMethod is null ?
+                                !IsValidOverrideReturnType(overridingProperty, overridingMemberType, overriddenMemberType, diagnostics) :
+                                !overridingMemberType.Equals(overriddenMemberType, TypeCompareKind.AllIgnoreOptions))
```

Comment:
>overriddenMemberType [](start = 61, length = 20)

It feels like this isn't always the right type to use. For example:
```
class A
{
     public virtual A P {get; set;}
}

class B : A
{
     public override B P {get => null;}
}

class C : B
{
     public override B P {get; set;}
}
```

It looks like the check will succeed for ```C.P```, but the override is not valid because we cannot override the setter this way.

Reply:
There is a separate check for the accessors which will fail.

---
In reply to: [416908389](https://github.com/dotnet/roslyn/pull/43673#discussion_r416908389) [](ancestors = 416908389)